react/ShippingHeader.tsx

```diff
@@ -3,17 +3,26 @@ import { ButtonPlain } from 'vtex.styleguide'
 import { PlaceDetails } from 'vtex.place-components'
 import { FormattedMessage } from 'react-intl'
 import { AddressContext } from 'vtex.address-context'
+import { OrderForm } from 'vtex.order-manager'
 
 interface Props {
   onEditAddress?: () => void
 }
 
 const { useAddressContext } = AddressContext
+const { useOrderForm } = OrderForm
 
 const ShippingHeader: React.VFC<Props> = ({ onEditAddress }) => {
   const { address, invalidFields } = useAddressContext()
+  const {
+    orderForm: { canEditData },
+  } = useOrderForm()
 
-  if (!address || invalidFields.includes('postalCode')) {
+  if (
+    !address ||
+    ((canEditData || address?.isDisposable) &&
+      invalidFields.includes('postalCode'))
```

Comment:
nit: IMO, this looks hard to read... do you think we can rewrite this in a clearer way? Maybe by using a named function, idk.

Reply:
I could extract the `canEditData || address?.isDisposable` to a variable named `isAddressEditable`, wdyt?

Reply:
lgtm